public/app/core/utils/richHistory.ts

```diff
@@ -118,6 +120,27 @@ export async function deleteQueryInRichHistory(id: string) {
   }
 }
 
+export async function migrateQueryHistoryFromLocalStorage(): Promise<boolean> {
+  const richHistoryLocalStorage = new RichHistoryLocalStorage();
+  const richHistoryRemoteStorage = new RichHistoryRemoteStorage();
+
+  try {
+    const richHistory: RichHistoryQuery[] = await richHistoryLocalStorage.getRichHistory({
+      datasourceFilters: [],
+      from: 0,
+      search: '',
+      sortOrder: SortOrder.Descending,
+      starred: false,
+      to: 14,
+    });
+    await richHistoryRemoteStorage.migrate(richHistory);
+    return true;
+  } catch (error) {
+    dispatch(notifyApp(createWarningNotification(`Query history migration failed. ${error.message}`)));
```

Comment:
Not sure if this is scope creep, but maybe we would want to track failures in analytics?